lib/pipeline.js

```diff
@@ -852,6 +853,128 @@ class PipelineModel extends BaseModel {
         return Promise.allSettled([...toCreateOrUpdate, ...toDeactivate]);
     }
 
+    /**
+     * Converts the simplified stages into a more consistent format
+     *
+     * This is because the user can provide the stage information as:
+     *  - {"name": { "jobs": ["job1", "job2", "job3"], "description": "Description" },
+     *     { "name2": { "jobs": ["job4", "job5"] } }
+     *
+     * We will convert it to a more standard format:
+     *  - [{ "name": "name", "jobs": [1, 2, 3], "pipelineId": 123, "description": "value" },
+     *     { "name": "name2", "jobs": [4, 5], "pipelineId": 123 }]
+     * @method convertStages
+     * @param  {Object}     config              config
+     * @param  {Object}     config.jobs         Jobs
+     * @param  {Object}     config.pipeline     Pipeline
+     * @param  {Object}     config.stages       Pipeline stages
+     * @return {Array}                New array with stages after up-converting
+     */
+    async _convertStages({ pipelineId, stages, jobs, pipelineJobs }) {
```

Comment:
What is the difference bw `jobs` and `pipelineJobs`?
We can remove `jobs` if it is redundant.
```suggestion
     * @param  {Object}     config.pipelineJobs         Jobs
     * @param  {Object}     config.pipeline                 Pipeline
     * @param  {Object}     config.stages                   Pipeline stages
     * @return {Array}                New array with stages after up-converting
     */
    async _convertStages({ pipelineId, stages, pipelineJobs }) {
```

Reply:
removed